src/Compute/Compute/VirtualMachine/Config/AddAzureVMNetworkInterfaceConfigurationCommand.cs

```diff
@@ -0,0 +1,140 @@
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.Azure.Management.Internal.Network.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    /// <summary>
+    /// Setup the network interface.
+    /// </summary>
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VMNetworkInterfaceConfiguration"), OutputType(typeof(PSVirtualMachine))]
+    public class AddAzureVMNetworkInterfaceConfigurationCommand : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+        [Alias("VMProfile")]
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMProfile)]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualMachine VM { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 1,
+           ValueFromPipelineByPropertyName = true)]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true)]
+        public bool? Primary { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true)]
+        [PSArgumentCompleter("Delete", "Detach")]
+        public string DeleteOption { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 4,
+           ValueFromPipelineByPropertyName = true)]
+        public string NetworkApiVersion { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 5,
+            ValueFromPipelineByPropertyName = true)]
+        public VirtualMachineNetworkInterfaceIPConfiguration[] IpConfiguration { get; set; }
+
+        [Parameter(
+            Mandatory = false)]
+        public SwitchParameter EnableAcceleratedNetworking { get; set; }
+
+        [Parameter(
+            Mandatory = false)]
+        public SwitchParameter EnableIPForwarding { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string NetworkSecurityGroupId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        [Alias("DnsServer")]
+        public string[] DnsSettingsDnsServer { get; set; }
+
+
+        public override void ExecuteCmdlet()
+        {
+            var networkProfile = this.VM.NetworkProfile;
+
+            if (networkProfile == null)
+            {
+                networkProfile = new NetworkProfile();
+            }
+
+            if (this.IsParameterBound(c=> this.NetworkApiVersion))
+            {
+                networkProfile.NetworkApiVersion = this.NetworkApiVersion;
```

Comment:
What happens if this is not set?
NetworkApiVersion left null?

We need to have a default API version here

Reply:
I guess this being an add that can make sense.

But we want to have some default set.

So like, if the value is null set to default, if not leave as is?

Providing netowrkAPiVersion on every call seems like a pain.